dsgrid/dataset/dataset_schema_handler_base.py

```diff
@@ -1,17 +1,14 @@
 import abc
 import logging
 import os
-from collections import defaultdict
 from typing import List
 
-import pyspark.sql.functions as F
-
 from dsgrid.config.dataset_config import DatasetConfig
 from dsgrid.config.simple_models import DatasetSimpleModel
 from dsgrid.dimension.base_models import DimensionType
-from dsgrid.exceptions import DSGInvalidDataset, DSGInvalidDimensionMapping, DSGInvalidField
+from dsgrid.exceptions import DSGInvalidDataset
 from dsgrid.dimension.time import TimeDimensionType
-from dsgrid.utils.spark import models_to_dataframe, check_for_nulls
+from dsgrid.utils.dataset import map_and_reduce_non_pivot_dimension, map_and_reduce_pivot_dimension
```

Comment:
How about `pivoted` and `stacked` dimensions?

Reply:
Sure. I'll make the change at the end of the PR. It will probably change lots of files.